File: app/src/main/java/lee/jaebeom/WikiPortal/java/SplashActivity.java
/*
package lee.jaebeom.WikiPortal;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.preference.PreferenceManager;

import java.util.Map;

import lee.jaebeom.WikiPortal.main.MainActivity;
import lee.jaebeom.WikiPortal.setting.SettingActivity;
import lee.jaebeom.WikiPortal.wiki.WikiData;
import lee.jaebeom.WikiPortal.wiki.WikiModel;


public class SplashActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        int wikis = 0;
        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(this);
        for (Map.Entry<String, WikiModel> entry: WikiData.wikiURL.entrySet()){
            entry.getValue().setUse(sharedPref.getBoolean(entry.getKey(), false));
            String lang = sharedPref.getString(entry.getKey() + "_language_setting", WikiData.korean);
            entry.getValue().setLanguage(lang);
            if (entry.getKey().equals(WikiData.Wikipedia)){
                entry.getValue().setChangeURLLanguage(lang);
            }
            if (entry.getValue().isUse()){
                wikis++;
            }
        }
        Intent activityIntent;
        if (wikis == 0){
            activityIntent = new Intent(this, SettingActivity.class);
        }else{
            activityIntent = new Intent(this, MainActivity.class);
        }
        startActivity(activityIntent);
        finish();

        overridePendingTransition(0, 0);
    }
}
*/

File: app/src/main/java/lee/jaebeom/WikiPortal/SplashActivity.kt
package lee.jaebeom.WikiPortal

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.preference.PreferenceManager
import lee.jaebeom.WikiPortal.main.MainActivity
import lee.jaebeom.WikiPortal.setting.SettingActivity
import lee.jaebeom.WikiPortal.util.SharedPreference
import lee.jaebeom.WikiPortal.wiki.WikiData
import lee.jaebeom.WikiPortal.wiki.WikiModel

/**
 * Created by leejaebeom on 2017. 11. 9..
 */
class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        var wikis = 0
        var sharedPref : SharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)

        WikiData.clearUseWikis()
        for (entry : Map.Entry<String, WikiModel> in WikiData.wikiURL.entries){
            entry.value.isUse = sharedPref.getBoolean(entry.key, false) //사용 하는 위키
            var lang : String  = sharedPref.getString(entry.key+ "_language_setting", WikiData.korean)  //언어 설정
            entry.value.language = lang
            if (entry.value.equals(WikiData.Wikipedia)){
                entry.value.setChangeURLLanguage(lang)  //언어에 따른 위키피디아 URL 변경
            }

            if (entry.value.isUse){
                WikiData.useWikis.add(entry.value)
                wikis++
            }

            entry.value.sequence = sharedPref.getInt("${SharedPreference.keySequence}_${entry.key}", 0)
        }

//        lateinit var activityIntent : Intent //1.2부터 가능
        var activityIntent : Intent?
        if (wikis == 0){
            activityIntent = Intent(this, SettingActivity::class.java)
        }else{
            activityIntent = Intent(this, MainActivity::class.java)
        }

        startActivity(activityIntent)
        finish()


        overridePendingTransition(0, 0)
    }
}